### PR TITLE
feat: New icons

### PR DESCRIPTION
**Pull request by hasecilu**
New icons for a small variety of extensions:

- Blender icon for `.blend` files.
- Godot icon for `.godot`, `.gd`, `.tres` and `.tscn` files. ~~**Note**: there are a lot of ASCII text file extensions used on godot, are you interested on all of these extensions to be added?~~
- G icon for `.gcode` files. These files are used on 3D printers and CNC machines.
- Add `.elf` files to `BINARY` category.
- Complete the missing extension from OpenDocument spec: `.fodt` for word processing (text) documents `.fods` for spreadsheets, `.fodp` for presentations `.fodg` for graphics and `.odf` for formulas.
- PlatformIO icon added to `platformio.ini` file ~~and `.pio` directory~~
- 0x icon for `.hex` files
- New `FILE_3D` category for 3D files: `.3mf`, `.obj`, `.ply` and `.stl` extensions. **Note**: This category is very extensive (+50), there are 3D files for manufacturing, for CAD exchange data, for AR/VR/videogames/movies, clouds of points, etc. Should be a differentiation?
- Square wave icon to extensions for logic analyzer sessions: `.sal` is for Saleae,`.sr` is for Sigrok/Pulseview.

```shell
# Command to recreate files
touch benchy.3mf BOM.fods bootloader.hex firmware.elf game.godot game_script.gd game_scene.tscn game_data.tscn lagrangian.odf lcd_comm.sal logo.fodg model.obj model.ply monkey.blend ocarina.stl platformio.ini sign.gcode slides.fodp spi_comm.sr writing.fodt 3d_asset.fbx
```

image

### CODE REVIEW

**Review by PThorpe92 — APPROVED** [review]
All fine additions :+1:  Thanks for the PR

Can you rebase to one conventional commit?

**Review by cafkafk — APPROVED** [review]
integration tests pass and this seem okay (when rebased as pthorpe suggests)

**Comment by cafkafk** [comment]
> there are a lot of ASCII text file extensions used on godot, are you interested on all of these extensions to be added?

> * This category is very extensive (+50), there are 3D files for manufacturing, for CAD exchange data, for AR/VR/videogames/movies, clouds of points, etc. Should be a differentiation?

You might wanna open an issue to discuss this, could always be a follow up PR

File: src/output/icons.rs [hunk 1]
```diff
@@ -120,6 +121,7 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     ".git"                => Icons::FOLDER_GIT,     // 
     ".github"             => Icons::FOLDER_GITHUB,  // 
     ".npm"                => Icons::FOLDER_NPM,     // 
+    ".pio"                => '\u{e682}',            // 
```
Comment (cfxegbert) [inline src/output/icons.rs:124]: Please make this a constant

File: src/output/icons.rs [hunk 2]
```diff
@@ -430,11 +436,16 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "flac"           => Icons::AUDIO,            // 
     "flv"            => Icons::VIDEO,            // 
     "fnt"            => Icons::FONT,             // 
+    "fodg"           => Icons::IMAGE,            // 
```
Comment (cfxegbert) [inline src/output/icons.rs:439]: Should be Icons::VECTOR.  Also add to filetypes.rs as a FileType::Image

File: src/output/icons.rs [hunk 3]
```diff
@@ -560,6 +573,9 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "nix"            => '\u{f313}',              // 
     "node"           => Icons::NODEJS,           // 
     "o"              => Icons::BINARY,           // 
+    "obj"            => Icons::FILE_3D,          // 󰆧
+    "odf"            => '\u{f0784}',             // 󰞄
+    "odg"            => Icons::IMAGE,            // 
```
Comment (cfxegbert) [inline src/output/icons.rs:578]: Same comment as above about vector and add to images filetype.

File: src/output/icons.rs [hunk 4]
```diff
@@ -660,7 +677,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "sqlite3"        => '\u{e7c4}',              // 
     "srt"            => Icons::SUBTITLE,         // 󰨖
     "ssa"            => Icons::SUBTITLE,         // 󰨖
-    "stl"            => Icons::IMAGE,            // 
+    "stl"            => Icons::FILE_3D,          // 󰆧
```
Comment (cfxegbert) [inline src/output/icons.rs:680]: Remove this from filetype.rs as an image type

**Review by cfxegbert — CHANGES_REQUESTED** [review]
A few minor changes

File: src/output/icons.rs [hunk 5]
```diff
@@ -414,6 +419,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "ejs"            => '\u{e618}',              // 
     "el"             => Icons::EMACS,            // 
     "elc"            => Icons::EMACS,            // 
+    "elf"            => Icons::BINARY,           // 
```
Comment (cfxegbert) [inline src/output/icons.rs:422]: Add to filetypes.rs as a FileType::Compiled

File: src/output/icons.rs (hunk 1 above)
Comment (cfxegbert) [reply src/output/icons.rs:124]: Actually just realized this is a directory.  We are trying to use only folder icons except for a few exceptions for things like Desktop, Photos, Downloads, ...

File: src/output/icons.rs (hunk 5 above)
Comment (hasecilu) [reply src/output/icons.rs:421]: All done

File: src/output/icons.rs (hunk 4 above)
Comment (cfxegbert) [reply src/output/icons.rs:681]: Can you please remove
```rust
"stl"        => FileType::Image,
```
from filetype.rs

**Comment by hasecilu** [comment]
@cfxegbert Hi, I solved the conflict with `main`. About the requested changes, they were done a month ago, is there anything missing?

**Comment by PThorpe92** [comment]
> @cfxegbert Hi, I solved the conflict with `main`. About the requested changes, they were done a month ago, is there anything missing?

This all looks great :+1: I will wait for him to respond to double check because he has put a lot of work into the current icon system. I would approve it but github `review` button hasn't been working for the last few hours.

Thanks for the contribution <3

**Comment by hasecilu** [comment]
I would say this PR is ready (I'm not adding nothing more here), any new icons I'll be adding it to #680

**Review by cafkafk — APPROVED** [review]
I'm okay with these as they are, at least the filetypes I'm familiar with.

I'll leave this on automerge for now, so it should merge when @cfxegbert approves it.

**Review by cafkafk — CHANGES_REQUESTED** [review]
Ohh, one oversight on my part, could I ask you to squash your commits? Ideally, with the commit summaries you squash in the commit description.

**Comment by hasecilu** [comment]
> Ohh, one oversight on my part, could I ask you to squash your commits? Ideally, with the commit summaries you squash in the commit description.

Done, a single commit.

**Review by cafkafk — CHANGES_REQUESTED** [review]
Hey, back from vacation, thanks for squashing these. Could you fix the merge conflict?

**Comment by hasecilu** [comment]
Sure, it's done.

**Comment by hasecilu** [comment]
@cfxegbert any thoughts?

**Review by cafkafk — CHANGES_REQUESTED** [review]
The merge commits here shouldn't be included, updates to the latest branch should be done with rebase to not include unnecessary merges in the commit history.

**Review by cafkafk — APPROVED** [review]
Thanks, this LGTM now, cool stuff :+1:

**Pull request merged**